nexus/db-model/src/disk.rs

```diff
@@ -44,6 +44,16 @@ pub struct Disk {
     #[diesel(embed)]
     pub runtime_state: DiskRuntimeState,
 
+    /// The PCI slot (within the bank of slots reserved to disks) to which this
+    /// disk should be attached if its attached instance is started, or None
+    /// if there is no such assignment.
+    ///
+    /// Slot assignments are managed entirely in Nexus and aren't modified by
+    /// runtime state changes in the sled agent, so this field is part of the
+    /// "main" disk struct and not the runtime state (even though the attachment
+    /// state and slot assignment will often change together).
+    pub slot: Option<i16>,
```

Comment:
IIRC, there are a few helper types for creating unsigned integers out of SQL signed types. `SqlU32` I think is a good example. That might be useful here, since this is basically a `u8` it seems.

Reply:
Great suggestion, thanks! Implemented (as you've seen) in f002264.